### ISSUE TITLE
I can't find the Classifier

### ISSUE THREAD

**Issue by ndhuynh02** [YubaoZhao/ECG-Chat#6]
Hi, I am looking at the code for finetuning with LoRa, and I am a little bit confused. 
As illustrated in Figure 2 of the paper, there is a classifier that predicts the probability for each diagnosis. However in the paper and also in the [finetune file](https://github.com/YubaoZhao/ECG-Chat/blob/master/llava/llm_finetune_lora.sh), it says that the generated report data from MIMIC-IV-ECG is used. As far as I know, this dataset doesn't have specific number of classes, as well as clear class label for each data sample. 
Therefore, I am wondering how many classes is used for the classifier, and how each MIMIC's sample is labeled to those classes.
P/s: I am also having a difficult time finding where the classifier takes place in the code

**Comment by YubaoZhao** [comment]
Hi, 

Thank you for your interest in our project. The classifier is used during inference, so no related code is in the training files. We built the classifiers on the PTB-XL training dataset with labels to enhance the diagnosis ability.

**Comment by ndhuynh02** [comment]
Does this mean in order to be able to evaluate the model on PTB-XL dataset, the classifier is 1st trained, then integrated to ECG-Chat? If this is the case, I am wondering whether the ECG encoder is finetuned along, or just the classification head is updated

**Comment by YubaoZhao** [comment]
You're right. Only an extra classification head is trained, the ECG encoder is frozen.

**Comment by ndhuynh02** [comment]
As I go deeper into the implementation, I see that the target diagnostic data used to finetune LoRa are passed through the model as "ASSISTANT". Does this mean the goal of LLM is to only generate a report based on the known cardiac diseases from an ECG signal, and doesn't actually perform signal analysis and provide insight/diagnostic from the input signal?

**Comment by YubaoZhao** [comment]
The tokens after the "ASSISTANT" are the training target of the Supervised Fine-Tuning (SFT). So, the LLM is trained to generate the report based only on the ECG signal. However, due to the limitation of training data, the model is not as accurate as we expected. So we use the results from a classifier as a "hint" in the inference stage. Our model is at an early stage, we're also trying to construct a more diverse SFT dataset and make it able to understand more fine-grained ECG features beyond the diagnostic result.
